daliuge-engine/dlg/data/drops/memory.py

```diff
@@ -65,60 +77,69 @@ def parse_pydata(pd: Union[bytes, dict]) -> bytes:
 
     :returns a byte encoded value
     """
-    pd_dict = pd if isinstance(pd, dict) else {"value":pd, "type":"raw"}
-    pydata = pd_dict["value"]
-    logger.debug("pydata value provided: '%s' with type '%s'", pydata, type(pydata))
+    stypes = ["float", "int", "str", "list", "dict"]
+    if not isinstance(pd, dict):
+        # fix types when just a value is passed (tests)
```

Comment:
@awicenec we really should avoid putting any more code that helps us pass tests. I will raise another issue to address this separately.